lonestar/libdistbench/include/DistBench/Output.h

```diff
@@ -0,0 +1,21 @@
+#ifndef GALOIS_DISTBENCH_OUTPUT_H
+#define GALOIS_DISTBENCH_OUTPUT_H
+
+#include <string>
+#include <fstream>
+
+std::string makeOutputFilename(const std::string& outputDir);
+
+template <typename T>
+void writeOutput(const std::string& outputDir, const std::string& /*fieldName*/,
+                 T* values, size_t length) {
```

Comment:
Why not take std::vector for values?

Reply:
That was in my original API :D Going to pointers may be a little more future-looking than necessary at the moment, but the idea is that this data may come from a pure buffer (GPU memory, mmap'ed file, etc). And if we go to purely zero copy, it will be difficult for users to encapsulate their data as `std::vector`s.

Reply:
Makes sense. Agreed. 

We can avoid copy for GPU but I'm not sure if I have exposed an API for that. I'll check and let you know.